relayer/relays/execution/message/message.go

```diff
@@ -72,22 +50,25 @@ func (m *Message) Sync(ctx context.Context, eg *errgroup.Group) error {
 
 			lastSyncedBlockNumber := executionHeaderState.BlockNumber
 
-			if lastSyncedBlockNumber > 0 && lastSyncedBlockNumber != secondLastSyncedBlockNumber {
+			if lastSyncedBlockNumber > 0 && (lastSyncedBlockNumber > lastSyncedIncentivizedBlock || lastSyncedBlockNumber > lastSyncedBasicBlock) {
 				log.WithFields(log.Fields{
-					"blockNumber": lastSyncedBlockNumber,
+					"lastSyncedIncentivizedBlock": lastSyncedIncentivizedBlock,
+					"lastSyncedBasicBlock":        lastSyncedBasicBlock,
+					"blockNumber":                 lastSyncedBlockNumber,
 				}).Info("last synced execution block changed, fetching messages")
 
-				err = m.syncIncentivized(ctx, eg, secondLastSyncedBlockNumber+1, lastSyncedBlockNumber)
+				err = m.syncIncentivized(ctx, eg, lastSyncedIncentivizedBlock+1, lastSyncedBlockNumber)
 				if err != nil {
 					return fmt.Errorf("sync incentivized messages: %w", err)
 				}
 
-				err = m.syncBasic(ctx, eg, secondLastSyncedBlockNumber+1, lastSyncedBlockNumber)
+				err = m.syncBasic(ctx, eg, lastSyncedBasicBlock+1, lastSyncedBlockNumber)
```

Comment:
When only one channel is behind, the up-to-date channel will have its sync method called with a start block that's 1 greater than the end block. Do the `FilterMessage` implementations work when the start block is greater than the end block? Just double-checking, I expect they do

Reply:
Good question! I've added a check to make sure the start block if smaller than the end block. https://github.com/Snowfork/snowbridge/pull/721/commits/6b732cc193a24114dd1f9891b20e280f9c1cb895